backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/LinesRangeTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SonarLint Core - Commons
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.commons;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class LinesRangeTest {
+
+  @Test
+  void it_should_test_illegal_arguments() {
+    assertThrows(IllegalArgumentException.class, () -> new LinesRange(-1, 1));
+    assertThrows(IllegalArgumentException.class, () -> new LinesRange(10, 1));
+  }
+
+  @Test
+  void testEquals() {
+    var range = new LinesRange(0, 42);
+    assertThat(range).isEqualTo(range)
+      .isEqualTo(new LinesRange(0, 42))
+      .isNotEqualTo(new LinesRange(1, 42))
+      .isNotEqualTo(new LinesRange(0, 41))
+      .isNotEqualTo(null)
+      .isNotEqualTo(new StringBuffer());
+  }
+
+  @Test
+  void testHashCode() {
+    assertThat(new LinesRange(0, 42)).hasSameHashCodeAs(new LinesRange(0, 42).hashCode());
+  }
```

Comment:
You can remove these tests if you agree to remove the overridden methods.

Reply:
Addressed in [179fcba](https://github.com/SonarSource/sonarlint-core/pull/949/commits/179fcbab50d500abff10bd003bc56d684f5e4102)